tests/Unit/API/Site/Controllers/Ads/BudgetRecommendationControllerTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Site\Controllers\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Proxy as Middleware;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads\BudgetRecommendationController;
+use Automattic\WooCommerce\GoogleListingsAndAds\DB\Query\BudgetRecommendationQuery;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\RESTControllerUnitTest;
+use Automattic\WooCommerce\GoogleListingsAndAds\Vendor\League\ISO3166\ISO3166DataProvider;
+use Exception;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * Class BudgetRecommendationControllerTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Site\Controllers\Ads
+ *
+ * @property BudgetRecommendationQuery|MockObject $budget_recommendation_query
+ * @property ISO3166DataProvider|MockObject       $iso_provider;
+ * @property BudgetRecommendationController       $controller
+ */
+class BudgetRecommendationControllerTest extends RESTControllerUnitTest {
+
+	protected const ROUTE_BUDGET_RECOMMENDATION  = '/wc/gla/ads/campaigns/budget-recommendation';
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->budget_recommendation_query = $this->createMock( BudgetRecommendationQuery::class );
+		$this->budget_recommendation_query->method( 'where' )
+			->willReturn( $this->budget_recommendation_query );
+
+		$this->iso_provider = $this->createMock( ISO3166DataProvider::class );
+		$this->middleware = $this->createMock( Middleware::class );
+
+		$this->controller = new BudgetRecommendationController( $this->server, $this->budget_recommendation_query, $this->middleware );
+		$this->controller->register();
+		$this->controller->set_iso3166_provider( $this->iso_provider );
+	}
+
+	public function test_get_budget_recommendation() {
+		$budget_recommendation_params = [
+			'country_codes' => ['JP', 'TW', 'GB', 'US'],
+		];
+
+		$budget_recommendation_data = [
+			[
+				'country'           => 'US',
+				'daily_budget_low'  => '330',
+				'daily_budget_high' => '930',
+			],
+			[
+				'country'           => 'GB',
+				'daily_budget_low'  => '245',
+				'daily_budget_high' => '625',
+			],
+			[
+				'country'           => 'TW',
+				'daily_budget_low'  => '95',
+				'daily_budget_high' => '255',
+			],
+			[
+				'country'           => 'JP',
+				'daily_budget_low'  => '110',
+				'daily_budget_high' => '320',
+			]
+		];
+
+		$expected_response_data = [
+			'currency'        => 'TWD',
+			'recommendations' => [
+				[
+					'country'           => 'US',
+					'daily_budget_low'  => 330.0,
+					'daily_budget_high' => 930.0,
```

Comment:
Even though we cast to float I seem to always be getting these values as 330 or 930, but never with the decimal added. Can we drop the decimals since it doesn't get returned that way in a response? It seems they are only added in the test since we are comparing with assertSame (which is strict about comparing the type), but if we cast them to int that's no longer needed.
This is what I'm seeing in the response even if they are floats, so I'd expect us to be testing with similar values:
```json
{
	"country": "US",
	"daily_budget_low": 10,
	"daily_budget_high": 25
}
```